regression/strings-smoke-tests/max_input_length/Test.java

```diff
@@ -0,0 +1,12 @@
+public class Test {
+
+    public static void main(String s) {
+	// This prevent anything from happening if string-max-length is smaller
```

Comment:
That comment is not clear about what should happen when `--string-max-length` is below 40. From a CBMC point of view I guess it's equivalent to unreachability. From a test-gen point of vue, it means no test will be generated for that block of code.
To check that the limit only affects a certain block, could we modify the test so that it looks like this:
```
public class Test {
    public static void main(String s, int i) {
        if (i==1) {
            String t = new String("0123456789012345678901234567890123456789");
            if(s.length() >= 30) {
                assert false;
            }
        } else {
            assert false;
        }
    }
}
```
and check that the first assertion is valid and the other one is failed?

Also `prevent` -> `prevents`.

Reply:
We should also have tests for input strings that are not function parameters.

Reply:
It appears that this `new String(...)` disables entire function, not only the block in which it is defined.

Reply:
OK, then we can just clarify the comment.